packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -198,6 +199,23 @@ impl Abigen {
         Ok(structs)
     }
 
+    // Checks whether the given type field is a Sway-native type.
+    // It's expected to come in as `"struct T"` or `"enum T"`.
+    // `T` is a Sway-native type if it matches exactly one of
+    // the reserved strings, such as "Address" or "ContractId".
+    fn is_sway_native_type(type_field: &str) -> bool {
+        let split: Vec<&str> = type_field.split_whitespace().collect();
+
+        if split.len() > 2 {
```

Comment:
Another way of writing this would have been:
```rust
match split.as_slice() {
   [_, CONTRACT_ID_SWAY_NATIVE_TYPE] | [_, ADDRESS_SWAY_NATIVE_TYPE] => true,
   _ => false
}
```